### PR TITLE
AI Features

### PR DESCRIPTION
**Pull request by steven-tey**
Sneak peek:

https://github.com/dubinc/dub/assets/28986134/fe85698e-212c-419a-b378-d7f31bc92ca6

### CODE REVIEW

File: apps/web/ui/modals/add-edit-link-modal/tags-section.tsx [hunk 1]
```diff
@@ -16,25 +24,82 @@ export default function TagsSection({
...
+    api: `/api/ai/completion?workspaceId=${workspaceId}`,
+    body: {
+      model: "claude-3-haiku-20240307",
+    },
+    onError: (error) => {
+      toast.error(error.message);
+    },
+    onFinish: (_, completion) => {
+      mutateWorkspace();
+      console.log({ completion });
```
Comment (TWilson023) [inline apps/web/ui/modals/add-edit-link-modal/tags-section.tsx]: Should we drop this `console.log`?

**Review by TWilson023 — APPROVED** [review]
Haven't tested locally but this looks great overall :+1:

File: apps/web/ui/modals/add-edit-link-modal/og-section.tsx [hunk 2]
```diff
@@ -28,8 +35,101 @@ export default function OGSection({
...
+        toast.custom(() => (
+          <UpgradeToProToast
+            title="You've exceeded your AI usage limit"
+            message={error.message}
+          />
+        ));
+      } else {
+        toast.error(error.message);
+      }
+    },
```
Comment (TWilson023) [inline apps/web/ui/modals/add-edit-link-modal/og-section.tsx:106]: Feels like this logic could maybe be split out, or even included somehow in `useCompletion`

File: apps/web/ui/modals/add-edit-link-modal/tags-section.tsx (hunk 1 above)
Comment (steven-tey) [reply apps/web/ui/modals/add-edit-link-modal/tags-section.tsx]: good catch

File: apps/web/ui/modals/add-edit-link-modal/og-section.tsx (hunk 2 above)
Comment (steven-tey) [reply apps/web/ui/modals/add-edit-link-modal/og-section.tsx:106]: agreed, this can definitely be optimized

Comment (steven-tey) [reply apps/web/ui/modals/add-edit-link-modal/og-section.tsx:106]: I also don't like that we're doing a `error.message.includes("Upgrade to Pro")` check, feels very fickle

**Pull request merged**